client/lib/analytics/record-enhanced-tracks-event.ts

```diff
@@ -0,0 +1,16 @@
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { withEnhancers } from 'calypso/state/utils';
+import type { Enhancer } from 'calypso/state/utils/with-enhancers';
+
+/**
+ * Record a tracks event with additional common properties using enhancers.
+ * @see client/state/analytics/README.md
+ */
+export default function recordEnhancedTracksEvent(
+	eventName: string,
+	eventData: any = {},
+	enhancers: Enhancer | Enhancer[] = []
+) {
+	const recordEvent = withEnhancers( recordTracksEvent, enhancers );
+	dispatch( recordEvent( eventName, eventData ) );
```

Comment:
I had trouble fixing this type error... 

```
client/lib/analytics/record-enhanced-tracks-event.ts(15,2): error TS2304: Cannot find name 'dispatch'.
```

I tried adding the following, but that didn't seem to work. The events didn't even fire when I made this change.

```diff
+import { useDispatch } from 'react-redux';
 import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { withEnhancers } from 'calypso/state/utils';
 import type { Enhancer } from 'calypso/state/utils/with-enhancers';
@@ -12,5 +13,7 @@ export default function recordEnhancedTracksEvent(
        enhancers: Enhancer | Enhancer[] = []
 ) {
        const recordEvent = withEnhancers( recordTracksEvent, enhancers );
+       const dispatch = useDispatch();
+
        dispatch( recordEvent( eventName, eventData ) );
 }
```

Reply:
`useDispatch` will only work from directly within React components (because it's a hook) and it seems we can't just `import { dispatch } from 'redux'`…

The main thing that strikes me here though is that the logic is different from `recordTracksEvent` ([source](https://github.com/Automattic/wp-calypso/blob/d71726ab9311383ae5f3674f6e4615a3dcf48082/packages/calypso-analytics/src/tracks.ts#L187)). That function pushes an array to `window._tkq`. Should we consider doing the same thing here as well?